Sources/audio-devices/AudioDevices.swift

```diff
@@ -9,87 +9,28 @@ struct AudioDevice: Hashable, Codable, Identifiable {
     case invalidVolumeValue
   }
 
-  enum TransportType: String, Codable {
-    case avb
-    case aggregate
-    case airplay
-    case autoaggregate
-    case bluetooth
-    case bluetoothle
-    case builtin
-    case displayport
-    case firewire
-    case hdmi
-    case pci
-    case thunderbolt
-    case usb
-    case virtual
-    case unknown
-
-    init(rawTransportType deviceTransportType: UInt32) {
-      switch deviceTransportType {
-        case kAudioDeviceTransportTypeAVB:
-          self = .avb
-        case kAudioDeviceTransportTypeAggregate:
-          self = .aggregate
-        case kAudioDeviceTransportTypeAirPlay:
-          self = .airplay
-        case kAudioDeviceTransportTypeAutoAggregate:
-          self = .autoaggregate
-        case kAudioDeviceTransportTypeBluetooth:
-          self = .bluetooth
-        case kAudioDeviceTransportTypeBluetoothLE:
-          self = .bluetoothle
-        case kAudioDeviceTransportTypeBuiltIn:
-          self = .builtin
-        case kAudioDeviceTransportTypeDisplayPort:
-          self = .displayport
-        case kAudioDeviceTransportTypeFireWire:
-          self = .firewire
-        case kAudioDeviceTransportTypeHDMI:
-          self = .hdmi
-        case kAudioDeviceTransportTypePCI:
-          self = .pci
-        case kAudioDeviceTransportTypeThunderbolt:
-          self = .thunderbolt
-        case kAudioDeviceTransportTypeUSB:
-          self = .usb
-        case kAudioDeviceTransportTypeVirtual:
-          self = .virtual
-        default:
-          self = .unknown
-      }
-    }
-  }
-
   let id: AudioDeviceID
   let name: String
   let uid: String
   let isInput: Bool
   let isOutput: Bool
```

Comment:
@karaggeorge I'm wondering if it would be nicer to make these two a single enum?

```swift
enum Type {
	case input
	case output
	case inputOutput
}
```

Can it be neither?

Reply:
Some prior art: https://github.com/rnine/AMCoreAudio/blob/07b528f26fdbe530b826007ca5635838153d10d6/Source/Public/Enums/Direction.swift

Reply:
So it seems a device can be both input and output, but not sure if it can be either.

```swift
enum Direction {
	case input
	case output
	case inputOutput
}
```

Reply:
Yeah I think a device can be both on or one of the two. I haven't seen one be neither, and I think that makes sense as it wouldn't be an audio device at that point.

If we made it into an enum, we'd have to also change the filtering logic when it asks for only input or output, the condition would have to be something like `{ $0.direction == input || $0.direction == inputOutput }`

Reply:
Yes, I don't think the enum would replace the boolean properties, it would just be a way to get a guaranteed correct state.

Reply:
But we don't have to do this now.